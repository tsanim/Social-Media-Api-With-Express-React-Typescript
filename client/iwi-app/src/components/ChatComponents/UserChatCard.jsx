import React from 'react';
import URI from '../../config/config';
import PropTypes from 'prop-types';

function UserChatCard(props) {
    return (
        <figure className={ props.isMine ? "float-right" : null }>
            <img src={`${URI}/feed/image/${props.onlineUser.imageId}`} alt="userImg" />
            <div className="names">
                <figcaption>{props.onlineUser.username}</figcaption>
            </div>
        </figure>
    )
}

UserChatCard.propTypes = {
    onlineUser: PropTypes.object,
    isMine: PropTypes.bool
}


export default UserChatCard;