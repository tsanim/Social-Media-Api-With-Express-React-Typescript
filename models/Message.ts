import mongoose, { Schema } from 'mongoose';
import IMessage from '../interfaces/models/Message.interface';

const messageSchema: Schema = new Schema({
    text: { type: Schema.Types.String },
    room: { type: Schema.Types.String },
    creator: { type: Schema.Types.ObjectId, ref: 'User' }
});

const Message = mongoose.model<IMessage>('Message', messageSchema);

export default Message;