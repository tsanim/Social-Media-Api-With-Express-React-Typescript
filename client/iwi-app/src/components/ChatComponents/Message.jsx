import React from 'react';
import PropTypes from 'prop-types';
import ReactEmoji from 'react-emoji';
import UserChatCard from './UserChatCard';

function Message(props) {
    return (
        <div className={`chat-message column ${props.isMine ? 'float-right' : null}`}>
            <UserChatCard
                onlineUser={props.message.creator}
                isMine={props.isMine}
            />
            <span className={props.isMine ? "mine" : null}>{ReactEmoji.emojify(props.message.text)}</span>
        </div>
    )
}

Message.propTypes = {
    isMine: PropTypes.bool,
    message: PropTypes.object
}


export default Message;


