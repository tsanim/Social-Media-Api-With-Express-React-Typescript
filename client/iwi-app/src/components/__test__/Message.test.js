import React from 'react';
import Message from '../ChatComponents/Message';
import { shallow } from 'enzyme';
import UserChatCard from '../ChatComponents/UserChatCard';

function messageComponentSetup() {
    const props = {
        isMine: true,
        message: { creator: {}, text: '' },
    }

    const enzymeWrapper = shallow(<Message {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('Message component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = messageComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div with class "chat-message"', () => {
        const { enzymeWrapper } = messageComponentSetup();

        expect(enzymeWrapper.find('div.chat-message')).toHaveLength(1);
    });

    it('should render a div tag', () => {
        const { enzymeWrapper } = messageComponentSetup();

        expect(enzymeWrapper.find('div')).toHaveLength(1);
    });

    it('should has child UserChatCard Component', () => {
        const { enzymeWrapper } = messageComponentSetup();

        expect(enzymeWrapper.find(UserChatCard)).toHaveLength(1);
    });
});