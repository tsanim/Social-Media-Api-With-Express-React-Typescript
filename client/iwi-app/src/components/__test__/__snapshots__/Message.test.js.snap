// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render initial layout 1`] = `
Array [
  <div
    className="chat-message column float-right"
  >
    <UserChatCard
      isMine={true}
      onlineUser={Object {}}
    />
    <span
      className="mine"
    />
  </div>,
]
`;
